cypress/e2e/gui/e2e.cy.js

```diff
@@ -0,0 +1,139 @@
+    cy.get('[data-testid="name"]').should('be.visible');
+  });
+
+  it('Abre e fecha o messenger', () => {
+    cy.get('[class^="Messenger_openCloseButton"]')
+      .should('have.attr', 'aria-label', 'Open messenger');
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('[class^="Messenger_openCloseButton"]')
+      .should('have.attr', 'aria-label', 'Close messenger');
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('[class^="Messenger_openCloseButton"]')
+      .should('have.attr', 'aria-label', 'Open messenger');
+  });
+
+  it('Garante que todos os campos do messenger são obrigatórios e que o primeiro está focado', () => {
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('button[type="submit"]').click();
+    cy.get('#messenger-name').should('have.attr', 'required');
+    cy.get('#email').should('have.attr', 'required');
+    cy.get('#message').should('have.attr', 'required');
+    cy.get('#messenger-name:first').should('be.focused');
+  })
+
+  it('Mostra e oculta uma mensagem de sucesso ao enviar o formulário do messenger com sucesso', () => {
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.clock()
+    cy.get('#messenger-name').type('joao');
+    cy.get('#email').type('joao@teste.com');
+    cy.get('#message').type('Testando');
+    cy.get('[class^="Messenger_sendButton"]').click();
+    cy.contains('Your message has been sent').should('be.visible')
+    cy.tick(3000)
+    cy.contains('Your message has been sent').should('not.exist')
+  })
+
+  it('Limpa todos os campos do formulário do messenger ao preenchê-los, fechar o messenger e abri-lo novamente', () => {
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('#messenger-name').type('joao');
+    cy.get('#email').type('joao@teste.com');
+    cy.get('#message').type('Testando');
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('[class^="Messenger_openCloseButton"]').click();
+    cy.get('#messenger-name').should('contain', '');
+    cy.get('#email').should('contain', '');
+    cy.get('#message').should('contain', '');
+  })
+
+  it('Mostra as colunas Nome da Empresa e Ação, e oculta as colunas ID, Indústria, Número de Funcionários e Tamanho em um viewport móvel', () => {
+    cy.viewport('iphone-x');
+    cy.contains('th', 'Company name').should('be.visible');
+    cy.contains('th', 'Action').should('be.visible');
+    cy.contains('th', 'ID').should('not.be.visible');
+    cy.contains('th', 'Number of employees').should('not.be.visible');
+    cy.contains('th', 'Size').should('not.be.visible');
+  })
+
+  it('Não encontra problemas de acessibilidade no modo claro na tabela de clientes', () => {
+    cy.checkA11y();
+  })
+
+  it('Não encontra problemas de acessibilidade no modo escuro na tabela de clientes', () => {
+    cy.get('[class^="ThemeToggle_button"]').click();
+    cy.checkA11y();
```

Comment:
Recomendo garantir que o tema mudou antes de fazer a verificação de acessibilidade.

Assim:

`cy.get('[data-theme="dark"]').should('exist')`

O mesmo vale para outros cenários.